lib/src/autosnapshotting/model.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:path/path.dart' as path;
+
+/// A record of a taken snapshot.
+class SnapshotRecord {
```

Comment:
maybe `SnapshotMetadata` since "record" is now a loaded term in Dart? Or just `Snapshot`?